packages/framework/esm-extensions/src/extensions.ts

```diff
@@ -357,15 +363,18 @@ function getAssignedExtensionsFromSlotData(
         }
       }
 
-      const displayConditionExpression = extensionConfig?.['Display conditions']?.expression ?? null;
-      if (displayConditionExpression !== null) {
+      const displayConditionExpression =
+        extensionConfig?.['Display conditions']?.expression ?? extension.expression ?? null;
```

Comment:
@CynthiaKamau So, because you provide an implicit config value of `""` for the `expression` under `"Display conditions"`, this statement will never resolve expressions defined via extension def in the routes file (`extension.expression`). This is because `extensionConfig?.['Display conditions']?.expression` will never be `null` or `undefined`. You should instead:

```suggestion
        extensionConfig?.['Display conditions']?.expression || extension.expression || null;
```

Reply:
@CynthiaKamau have you tested this out at runtime to confirm that it works extension config? If so, can you show me a sample config?